gneiss/_summary.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+import pandas as pd
+
+
+class RegressionResults():
+    """
+    Summary object for storing regression results.
+    """
+    def __init__(self, stat_results,
+                 feature_names=None,
+                 basis=None):
+        """ Reorganizes statsmodels regression modules.
+
+        Accepts a list of statsmodels RegressionResults objects
+        and performs some addition summary statistics.
+
+        Parameters
+        ----------
+        stat_results : list, sm.RegressionResults
+            List of RegressionResults objects.
+        feature_names : array_like, str
+            List of original names for features.
+        basis : np.array, optional
+            Orthonormal basis in the Aitchison simplex.
+            If this is not specified, then `project` cannot
+            be enabled in `coefficients` or `predict`.
+        """
+        self.feature_names = feature_names
+        self.basis = basis
+        self.results = stat_results
+
+        # obtain pvalues
+        pvals = pd.DataFrame()
+        for i in range(len(self.results)):
+            p = self.results[i].pvalues
+            p.name = self.results[i].model.endog_names
+            pvals = pvals.append(p)
+        self.pvalues = pvals
+
+        # calculate the overall R2 value
+        sse = sum([r.ssr for r in self.results])
```

Comment:
Here's some timings.  Some boiler plate code is below so that you can reproduce this.

```python
In [1]: import pandas as pd
In [2]: from gneiss._summary import RegressionResults
In [3]: import statsmodels.formula.api as smf
In [6]: data = pd.DataFrame([[1, 3, 4, 5, 2, 3, 4],
   ...:                       list(range(1, 8)),
   ...:                       [1, 3, 2, 4, 3, 5, 4]],
   ...:                      columns=['s1', 's2', 's3', 's4',
   ...:                               's5', 's6', 's7'],
   ...:                      index=['Y1', 'Y2', 'X']).T
In [10]: model1 = smf.ols(formula="Y1 ~ X", data=data)
In [11]: model2 = smf.ols(formula="Y2 ~ X", data=data)
In [12]: results = [model1.fit(), model2.fit()]
```

When I run my code, here's the following time
```python
In [15]: %timeit RegressionResults(results)
100 loops, best of 3: 2.75 ms per loop
```

Your suggested needed a little tweaking (i.e. `ssr` and `sse` shouldn't be lists, they should be number).  I modified that code as follows

```python
        sse = 0
        ssr = 0
        for i in range(len(self.results)):
            p = self.results[i].pvalues
            p.name = self.results[i].model.endog_names
            pvals = pvals.append(p)
            ssr += self.results[i].ssr
            sse += self.results[i].ess
```
When I run that code, I get the following timings

```python
In [8]: %timeit RegressionResults(results)
100 loops, best of 3: 5.31 ms per loop
```
So about twice as slow.

What's probably going on is there is that the list comprehension + np.sum is vectorized, allowing for more compiler optimizations.  Plus it is much more readable IMHO.

Do you think this is still ok to keep as is?